DoublyLinkedList/doubly_linked_list.h

```diff
@@ -0,0 +1,287 @@
+using namespace std;
+// Node Creation
+class node
+{
+public:
+    int data;
+    node *next;
+    node *prev;
+};
+
+class doubly_linked_list : private node
+{ //class
+    node *head=nullptr;
+    node *tail=nullptr;
+    int length = 0;
+
+public:
+    //for creating new node
+    node* create_node(node *x, int value, node *y);
+
+    // insertion of value at front & back
+    void push_front(int value);
+    void push_back(int value);
+
+    //for inserting at location
+    void insert(int value, int location);
+
+    //for removing values at front and back
+    int remove_front();
+    int remove_back();
+
+    //for removing at a location and from a range
+    void remove(int location);
+    void remove(int start, int end);
+
+    //for finding the size
+    size_t size();
+
+    //check whether empty or not
+    bool is_empty();
+
+    //Printing of values
+    void print_values();
+
+    //to clear all the elements in the list
+    void clear();
+
+};
+//for  creating new node
+node* doubly_linked_list::create_node(node *x, int value, node *y){
+    node *new_node = new node();
+    new_node->data = value;
+    new_node->prev = x;
+    new_node->next = y;
+    return new_node;
+};
+
+//adding element at the front
+void doubly_linked_list::push_front(int value){
+    //new node creation
+    node * new_node = create_node(nullptr, value, head);
+
+    //if head node is not empty then prev of head node will be assigned to the new node
+    if (head != nullptr)
+    {
+        head->prev = new_node; 
+    }
+    //if the node is empty then both head and tail will point to the new node
+    if (head == nullptr)
+    {
+        tail = new_node;
+    }
+    //now the new node becomes the first node
+    head = new_node; 
+    length++;
+}
+//add the element at the back
+void doubly_linked_list::push_back(int value) {
+    //new node creation
+    node *new_node = create_node(tail,value,nullptr);
+    //if the list is not empty then tail node is not empty then next of tail node will be assigned to the new node
+    if (tail!=nullptr)
+    {
+        tail->next = new_node; 
+    }
+    if (tail==nullptr)
+    {
+        head = new_node;
+    }
+    //now the new node becomes the last node
+    tail = new_node; 
+    length++;
+}
+//for inserting at a location
+void doubly_linked_list::insert(int value, int location){
+    if(location > length || location <0)
+    {
+        cout << "Invalid Location specified to Insert"<<endl;
+    }
+    else
+    {
+        if (location == 0){
+            push_front(value);
+            return;
+        }
+        else if (location == length){
+            push_back(value);
+            return;
+        }
+        else
+        {
+                //created temporary nodes which will point to previous and next node
+            node *temp1 = head; 
+            node *temp2;
+            int count = 1;
+                //iterating to insert at a position 
+            while (count != location) {
```

Comment:
move this to a separate function that takes position and returns the node at the position. 

make it a private method.